### ISSUE TITLE
Can't switch node versions from 10.0.0

### ISSUE THREAD

**Issue by tsauvajon** [coreybutler/nvm-windows#350]
### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL - see wiki for details)
- [ ] Windows 8
- [ ] Windows 8.1
- [ ] Windows 10
- [ ] Windows 10 IoT Core
- [x] Windows Server 2012
- [ ] Windows Server 2012 R2

- [ ] Windows Server 2016
- [x] My Windows installation is non-English.

### I'm using NVM4W version:

- [x] 1.1.6
- [x] 1.1.5
- [ ] 1.1.4
- [ ] 1.1.3
- [ ] 1.1.2
- [ ] 1.1.1
- [ ] Older
- [ ] OTHER (Please Specify)

### I have already...

- [x] read the README to be aware of npm gotchas & antivirus issues.
- [x] reviewed the [wiki](https://github.com/coreybutler/nvm-windows/wiki) to make sure my issue hasn't already been resolved.
- [x] verified I'm using an account with administrative privileges.
- [x] searched the issues (open and closed) to make sure this isn't a duplicate.
- [x] made sure this isn't a question about how to use NVM for Windows, since [gitter](https://gitter.im/coreybutler/nvm-windows?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) is used for questions and comments.

### My issue is related to (check only those which apply):

- [ ] settings.txt
- [ ] proxy support (Have you tried version 1.1.0+?)
- [ ] 32 or 64 bit support (Have you tried version 1.1.3+?)
- [ ] Character escaping (Have you tried version 1.1.6+?)
- [x] A standard shell environment (terminal/powershell)
- [ ] A non-standard shell environment (Cmder, Hyper, Cygwin, git)

### Expected Behavior

`nvm use 9.11.1` should change from version `10.0.0` to `9.11.1`

### Actual Behavior

```
>node -v
v10.0.0

>nvm use 9.11.1
Now using node v9.11.1 (64-bit)

>node -v
v10.0.0

>nvm version
1.1.6

>nvm list
  * 10.0.0 (Currently using 64-bit executable)
    9.11.1
```

It works correctly on my local machine (Windows 7) but doesn't work on my CI Server (Windows 2012), which is very problematic as it breaks some builds.
I've tried nvm versions `1.1.5` and `1.1.6`.
I had no problem switching from node version `9.11.1` to `10.0.0`, but it doesn't seem to work the other way around.

### Steps to reproduce the problem:

On a Windows 2012 Administration Command Prompt:
- `nvm install 10.0.0`
- `nvm use 10.0.0`
- `nvm install 9.11.1`
- `nvm use 9.11.1`

**Comment by tsauvajon** [comment]
I'm not sure why but after many tries it worked.

I didn't change anything, just started new administrator shells several times and at some point it changed the version (and the change was visible only by opening yet another shell).

I had the exact same issue on my 3 CI servers (Windows Server 2012), so I'm not sure if I should close the issue.

**Comment by Mohamed3on** [comment]
Can confirm this issue. Unable to downgrade to node 8.

**Comment by tsauvajon** [comment]
@mohamed3on check my second message for the solution if you're still stuck with the problem.

**Comment by Mohamed3on** [comment]
I restarted the shell and it still doesn't work. Also, I cannot uninstall v10 (it says it's not installed)
![image](https://user-images.githubusercontent.com/12295159/39710911-749298b0-521e-11e8-99aa-607d50b2f915.png)

**Comment by tsauvajon** [comment]
Can you check your environment variables? Look for `NVM_SYMLINK`

**Comment by jbergens** [comment]
I uninstalled nvm and then installed the new 1.1.7 version. Then I could install node 10.1.0 but that version downloaded and installed npm 5.6.0 which is too old. That may be another problem though.

**Comment by b-dur** [comment]
@jbergens good to know that you could install node 10.1.0.

I quickly looked into where the npm version comes from. NVM fetches the info from https://nodejs.org/dist/index.json and extract the versions. Can see in the list that node 10.1.0 is registered with npm 5.6.0.

However there is a problem ATM with upgrading npm locally.

**Comment by narve** [comment]
As far as I can see, these problems, in combination, is currently a showstopper. nvm can not be used to install the latest (10+) versions of node and npm. And npm can not be updated (easily) when installing node via nvm. Which indeed makes nvm unusable for lots of peope, e.g. me :) 

Does anyone know why https://nodejs.org/dist/index.json (and http://nodejs.org/dist/index.tab) features such old versions of npm? Is it a bug, or by design? 

And given that it is so, how can we fix it? Either make sure that node-installation comes with a recent npm, or make updating npm work?

**Comment by tsauvajon** [comment]
How to fix:

Open a new cmd shell and run
```
echo %NVM_SYMLINK%
where node
```

If node is located under the symlink location (e.g. in my case `NM_SYMLINK` = `C:\Program Files\nodejs` and node is located at `C:\Program Files\nodejs\node.exe`, it should be OK.
Just restart your computer and everything should work fine.

If nodejs is located somewhere else (eg: C:\nodejs\node.exe), just delete the containing "directory", which is actually just a symlink. In my case, I would open an explorer, navigate to C:\, right click on the nodejs symlink (the default icon is a folder icon with an arrow on it) and delete it. You can also do that by command line: `rmdir c:\nodejs` (or whatever your old symlink was - note: I haven't tested this). Now restart your computer, re install and use the desired node version with nvm and you should be good to go.

**Comment by narve** [comment]
@tsauvajon  thanks for replying. This doesn't work for me, if I understand you correct. 

```
C:\>echo %NVM_SYMLINK%
C:\Program Files\nodejs

C:\>where node
C:\Program Files\nodejs\node.exe

C:\>nvm list

  * 10.2.1 (Currently using 64-bit executable)
    10.1.0
    9.11.1

C:\>npm --version
npm WARN npm npm does not support Node.js v10.2.1
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
5.6.0

C:\>npm -g i npm
npm WARN npm npm does not support Node.js v10.2.1
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN npm npm does not support Node.js v10.2.1
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 4, 6, 7, 8, 9.
npm WARN npm You can find the latest version at https://nodejs.org/
npm ERR! path C:\Program Files\nodejs\npm.cmd
npm ERR! code EEXIST
npm ERR! Refusing to delete C:\Program Files\nodejs\npm.cmd: is outside C:\Program Files\nodejs\node_modules\npm and not a link
npm ERR! File exists: C:\Program Files\nodejs\npm.cmd
npm ERR! Move it away, and try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\narve\AppData\Roaming\npm-cache\_logs\2018-05-28T11_33_46_169Z-debug.log

C:\>
```

Did you have a different experience, or did I misunderstand?

**Comment by tsauvajon** [comment]
Oh actually your node installation is working correctly but npm won't install npm, I misunderstood your problem.

I had similar incompatibilities when upgrading to node 10.x, I had to roll back to 9.x.

I think this is a node / npm problem, not a nvm problem. Do you have a different experience when installing node 10 manually ? I haven't taken the time to investigate anything, I just went back to v9.x for now, with all these warnings (it looks like v10 isn't supported by npm / yarn yet for some reason).

**Comment by narve** [comment]
I think this is a node problem (node ships with incompatible npm version -- how is this possible!?), which in combination with the nvm problem (can't upgrade npm) makes nvm unusable for latest node versions. Is this the correct understanding? 

Last time I installed node 10+ manually I believe I was able to update npm, but I may remember wrong.

**Comment by tsauvajon** [comment]
After some research I've found a way to update npm. You have to move out of the default repository for some reason.

With nvm, the binaries will be installed at `%NVM_SYMLINK%`.

If you want to either:
- update npm
- install global packages with yarn
you will have to change where the binaries are installed.

Note that some packages don't work with every version of node !

Run the following:
```cmd
npm config set prefix "%appdata%/npm" -- or another folder of your choice
npm i -g npm yarn
yarn config set prefix "%appdata%/npm"
```

**Comment by narve** [comment]
Thanks! That worked for me :)

**Comment by lpreterite** [comment]
My situation is like this, the current version of node is 16, and the use of version 14 is invalid.

```sh
$ nvm current
v16.20.0

$ nvm use 14.18.2
Now using node v14.18.2 (64-bit)

$ nvm current
v16.20.0
```

It was later discovered that after reinstalling nvm-windows v1.1.11, there were two symbolic links.

```sh
$ where node
C:\nodejs\node.exe
C:\Program Files\nodejs\node.exe
```

After checking the file properties, it is found that the symbolic link of `C:\nodejs\` points to version 16, and the symbolic link of `C:\Program Files\nodejs\` points to version 14.

Later, after I deleted `C:\nodejs\node.exe`, the node version in the environment can be displayed normally.

```sh
$ nvm current
v14.18.2
```

**Comment by coreybutler** [comment]
I believe everything from the original issue has been resolved in newer versions (this was originally filed for NVM4W 1.1.5/6), so I am closing this. It would be best to open another issue if this persists in modern versions.

In NVM4W 1.1.11, there is a `nvm debug` command. It runs a check to determine whether NVM is in the correct location within your `PATH`. The case of having NVM4W install in a new location (essentially installed twice) was not one I thought of when adding the `debug` command, so I've added that to my backlog.

To avoid the situation described by @lpreterite, either make sure you use the same installation paths when you upgrade, or use the special upgrade tool packaged in the release (which only upgrades the release assets, not the environment variables or settings files).
